File: backend/routers/clients.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime, timezone
from utils.supabase_client import supabase

router = APIRouter(prefix="/api/clients", tags=["Clients"])

class ClientCreate(BaseModel):
    company: str
    email: str
    industry: Optional[str] = ""
    notes: Optional[str] = ""
    size: Optional[str] = ""

async def generate_client_id():
    # Use RPC or just select all and find max for now to keep it simple
    res = supabase.table("clients").select("client_id").execute()
    clients_list = res.data or []
    comps = [c.get("client_id", "") for c in clients_list if isinstance(c.get("client_id"), str)]
    valid_ids = [int(i.replace("FRIST", "")) for i in comps if i.startswith("FRIST") and i.replace("FRIST", "").isdigit()]
    next_id = max(valid_ids) + 1 if valid_ids else 1
    return f"FRIST{next_id:03d}"

@router.get("/")
async def get_clients():
    res = supabase.table("clients").select("*").order("created_at", desc=True).execute()
    return res.data or []

@router.get("/next-id")
async def get_next_id():
    nid = await generate_client_id()
    return {"next_id": nid}

@router.get("/{client_id}")
async def get_client(client_id: str):
    res = supabase.table("clients").select("*").eq("client_id", client_id).execute()
    if not res.data:
        raise HTTPException(status_code=404, detail="Client not found")
    return res.data[0]

@router.post("/", status_code=201)
async def create_client(req: ClientCreate):
    client_id = await generate_client_id()
    new_client = {
        "client_id": client_id,
        "company": (req.company or "").strip(),
        "industry": (req.industry or "").strip(),
        "email": (req.email or "").strip(),
        "notes": (req.notes or "").strip(),
        "size": (req.size or "").strip(),
        "created_at": datetime.now(timezone.utc).isoformat()
    }
    res = supabase.table("clients").insert(new_client).execute()
    if not res.data:
         raise HTTPException(status_code=500, detail="Failed to create client")
    return res.data[0]

@router.put("/{client_id}")
async def update_client(client_id: str, updates: dict):
    # Ensure client_id is not in updates or matches
    updates.pop("client_id", None)
    res = supabase.table("clients").update(updates).eq("client_id", client_id).execute()
    if not res.data:
        raise HTTPException(status_code=404, detail="Client not found")
    return res.data[0]

@router.delete("/{client_id}")
async def delete_client(client_id: str):
    res = supabase.table("clients").delete().eq("client_id", client_id).execute()
    if not res.data:
        raise HTTPException(status_code=404, detail="Client not found or deletion failed")
    return {"success": True}

File: backend/src/clients.js
const router = require('express').Router();
const store  = require('../middleware/store');
const { v4: uuid } = require('uuid');

function generateClientId(clients) {
  const existing = clients
    .map(c => c.client_id || '')
    .filter(id => /^FRIST\d{3,}$/.test(id))
    .map(id => parseInt(id.replace('FRIST', ''), 10))
    .filter(n => !isNaN(n));
  const next = existing.length > 0 ? Math.max(...existing) + 1 : 1;
  return 'FRIST' + String(next).padStart(3, '0');
}

/* GET /api/clients — list all */
router.get('/', (req, res) => {
  const clients = store.read('clients.json', []);
  res.json(clients);
});

/* GET /api/clients/next-id — get the next available client ID */
router.get('/next-id', (req, res) => {
  const clients = store.read('clients.json', []);
  res.json({ next_id: generateClientId(clients) });
});

/* GET /api/clients/:id — single client */
router.get('/:id', (req, res) => {
  const clients = store.read('clients.json', []);
  const client  = clients.find(c => c.client_id?.toLowerCase() === req.params.id.toLowerCase());
  if (!client) return res.status(404).json({ error: 'Client not found' });
  res.json(client);
});

/* POST /api/clients — create */
router.post('/', (req, res) => {
  const { company, industry, email, notes, size } = req.body;
  if (!company || !email) return res.status(400).json({ error: 'company and email required' });

  const clients   = store.read('clients.json', []);
  const client_id = generateClientId(clients);
  const newClient = {
    client_id,
    company: company.trim(),
    industry: industry?.trim() || '',
    email: email.trim(),
    notes: notes?.trim() || '',
    size: size?.trim() || '',
    createdAt: new Date().toISOString(),
  };
  clients.push(newClient);
  store.write('clients.json', clients);

  res.status(201).json(newClient);
});

/* PUT /api/clients/:id — update */
router.put('/:id', (req, res) => {
  const clients = store.read('clients.json', []);
  const idx     = clients.findIndex(c => c.client_id === req.params.id);
  if (idx < 0) return res.status(404).json({ error: 'Client not found' });

  clients[idx] = { ...clients[idx], ...req.body, client_id: req.params.id };
  store.write('clients.json', clients);
  res.json(clients[idx]);
});

/* DELETE /api/clients/:id — delete */
router.delete('/:id', (req, res) => {
  let clients = store.read('clients.json', []);
  const before = clients.length;
  clients      = clients.filter(c => c.client_id !== req.params.id);
  if (clients.length === before) return res.status(404).json({ error: 'Client not found' });
  store.write('clients.json', clients);
  res.json({ success: true });
});

module.exports = router;
